### PR TITLE
Responsive Site

### PR DESCRIPTION
**Pull request by OzzieOrca**
So I was thinking about the questions page and how it was not very mobile friendly both because the drag and drop didn't work and because nothing scaled down to the phone size. I was going to prototype it a little bit and see what ideas I had but it turned into much more than that with a new questions toolbar, pages and their blocks in panels and toolbars for each of those blocks.

image
image

I completely killed the drag functionality with my HTML and CSS rewrite although if this moves forward I intend to add it back. It was nice to not have to worry about it as I modified stuff and the current implementation didn't work on touch devices anyways.

I was looking at using one of these libraries for drag and drop since they work on touch devices:
http://jimliu.github.io/angular-ui-tree/index.html - Has great Angular bindings. I successfully used it to reorder the toolbar of new questions (although I haven't committed that yet)
http://rubaxa.github.io/Sortable/
http://touchpunch.furf.com/

What are your guys' initial thoughts? I probably should have discussed it first but I got a piece of it working and just kept going. I think it is more user friendly but I would like to hear feedback. Also, since I enabled responsiveness, some of the other pages and the header don't scale correctly in narrower viewports. That would take a bit to go through and fix but it shouldn't be too bad and should be done anyway if we want the site to be responsive.

This is NOT ready to merge, I just wanted to share it.

Also the delete page btn is a little big but I was trying to figure out what to do with it. Maybe I'll split it into a separate button group...

### CODE REVIEW

**Comment by OzzieOrca** [comment]
@adammeyer Can you take a look at this again when you get a chance? I think I completed my list of things that I still needed to do with this PR. I'm not sure what you think of the btn-info color (used on Preview Form and Edit below). It is using the `@lightblue` LESS color that was at the top of `style.less`
image

**Comment by adammeyer** [comment]
Looking good.  2 things:
1) Is the "Cru Profile" button clickable?  If not, remove the border the so it doesn't look like a button.
2) I think the color needs to be reduced on this page.  Try making the "Edit" and "Copy" buttons the same as the "Move" white.  Same with the "Preview Form" button.  Try it in white.

**Comment by OzzieOrca** [comment]
What does Cru Profile do exactly? I'm assuming it links the field(s) with some already existing info in a database. Is this explained to the admin when they are creating questions? Would this be commonly used or would most people leave it on default? Do admins need to quickly see whether the profile is enabled for each question? Is the action used enough to pull the button out of the edit section that expands? Currently it is just displaying whether it is enabled. I could make it clickable if that would be useful.

I made the rest of the buttons white. I feel like Edit should stand out a little bit more since it is used for accessing a lot of features. I kinda like the btn-warning color but it's not used anywhere else in the app currently.

**Comment by adammeyer** [comment]
It's supposed to save your answers you use in those fields, and have them automatically be set for any future events you register for.  But I've never seen it actually work.  I'll ask Ryan if the backed is even supporting it.

I don't think it needs to be displayed up top for everyone question.  Maybe if the user clicks "Edit" for that question.  Like how it is now: 
image
What do you think?

**Comment by OzzieOrca** [comment]
Does it just populate all blocks of the same type in the future?
I can put it wherever. If it's not used much I can put it back in the edit section. I think I moved it up cuz it was in the same row as delete and copy but now that I think about it there wasn't much of a reason when I moved it.

**Comment by adammeyer** [comment]
All blocks of the same type that have "Cru Profile" selected. But you can only have one Cru Profile per question per event.

**Comment by OzzieOrca** [comment]
Ok. So does putting it back in the edit section sound good? I was thinking about having that green text and icon be the label for the checkbox

**Comment by adammeyer** [comment]
Yes, sounds good.  I'm waiting to hear back from Ryan on it.  I don't think it even is functioning, so it might get removed.  I'll let you know when I hear back.

**Comment by adammeyer** [comment]
I like it.  I'm working on pushing this to stage now.

File: app/views/eventDetails/eventInformation.html [hunk 1]
```diff
@@ -1,39 +1,38 @@
...
         <option value="America/Halifax">Atlantic</option>
         <option value="America/Puerto_Rico">Atlantic Standard</option>
-        <option value="America/New_York">Eastern</option>
-        <option value="America/Chicago">Central</option>
-        <option value="America/Denver">Mountain</option>
-        <option value="America/Phoenix">Mountain Standard</option>
-        <option value="America/Los_Angeles">Pacific</option>
-        <option value="America/Anchorage">Alaska</option>
-        <option value="Pacific/Honolulu">Hawaii</option>
+      <option value="America/New_York">Eastern</option>
```
Comment (adammeyer) [inline app/views/eventDetails/eventInformation.html]: Is there a reason you removed the indents here?

File: app/views/modals/loginDialog.html [hunk 2]
```diff
@@ -4,9 +4,16 @@
...
+    <p class="lead spacing-below-xs">
+      Sign in with one of the following services:
+    </p>
+    <p>
+      <a class="btn btn-subtle btn-spacing" ng-href="{{apiUrl}}auth/facebook/authorization" title="Facebook"><img src="/img/signin-facebook.png"></a>
+      <a class="btn btn-subtle btn-spacing" ng-href="{{apiUrl}}auth/relay/login?logoutCallbackUrl={{apiUrl}}auth/relay/logout" title="Relay"><img src="/img/signin-relay.png"></a>
+    </p>
+    <p>
+      <span class="text-muted">or</span>
+      <a class="logo ng-scope" href="#/" ng-click="close()">Return to the Welcome Page</a>
```
Comment (adammeyer) [inline app/views/modals/loginDialog.html]: This should say "Go to Welcome Page" or something like that.  Most of the time the users will not becoming from there.

File: app/views/modals/paymentsModal.html [hunk 3]
```diff
@@ -1,324 +1,329 @@
...
+          </tbody>
+        </table>
+      </div>
+    </tab>
+    <tab heading="Add">
+      <div class="row form-group">
+        <div class="col-sm-7 stacked-spacing-col-sm">
+          <label>Type</label>
+          <select ng-model="newTransaction.paymentType" class="form-control">
+            <option value="" disabled selected>Choose one...</option>
```
Comment (adammeyer) [inline app/views/modals/paymentsModal.html]: Nice!

File: app/views/modals/paymentsModal.html [hunk 4]
```diff
@@ -1,324 +1,329 @@
...
+            </select>
           </div>
 
-          <div ng-show="newTransaction.paymentType === 'CHECK'">
-            <div class="form-group">
-              <label>Check Number</label>
-              <input ng-model="newTransaction.check.checkNumber" type="text" class="form-control" />
-            </div>
+          <div class="col-sm-3">
+            <label>Verification Code</label>
```
Comment (adammeyer) [inline app/views/modals/paymentsModal.html]: I think "Security Code" would be more common language.

File: app/index.html [hunk 5]
```diff
@@ -185,8 +187,8 @@ <h1 ng-bind="globalPage.title"></h1>
   </header>
   <section id="frontpagetitle">
     <div class="container">
-      <div class="row">
-        <div class="col-xs-10">
+      <div class="row spacing-above-md spacing-below-md">
+        <div class="col-sm-10 stacked-spacing-col-sm">
           <a ng-href="#/{{registerMode || 'register'}}/{{globalPage.confId}}/page/" ng-if="globalPage.confId">
```
Comment (adammeyer) [inline app/index.html]: A hyperlink underline is added when hovering over the event name.  That should be removed.

**Comment by adammeyer** [comment]
Looking great @OzzieOrca!  One idea I had: Could the event blocks on the dashboard be smaller?  Currently I have 7 events on mine, and it seems like a lot space. Maybe like the 2nd one?
shorter

What do you think?

File: app/views/eventDetails/eventInformation.html (hunk 1 above)
Comment (OzzieOrca) [inline app/views/eventDetails/eventInformation.html]: That probably happened in a merge... I'll fix it

**Comment by OzzieOrca** [comment]
I think I fixed all the issues above.

With the dashboard events, I reduced the padding. Here's what it looks like now
sm device:
image
xs device:
image

I could also switch it so the "3 Registered" button is in a row with the title:
sm device (reduces space for title):
image
xs device (I don't like how this breaks it up. I suppose it could somewhere else on xs devices):
image

**Comment by adammeyer** [comment]
I like it on the same row as the title.  Could you float it right so there isn't as much space between it and the title?

**Comment by OzzieOrca** [comment]
That actually works a lot better. Idk why I didn't try that before.
image

**Comment by adammeyer** [comment]
Looks great!

File: app/views/eventDetails/paymentOptions.html [hunk 6]
```diff
@@ -1,135 +1,150 @@
...
+  </div>
+</div>
 <div class="well" collapse="!conference.acceptCreditCards">
   <div class="form-group">
-    <div class="row">
-      <div class="col-xs-6">
-        <label>Payment Gateway:</label>
-        <select class="form-control" ng-model="conference.paymentGatewayType" ng-options="gateway.id as gateway.name for gateway in paymentGateways"></select>
-      </div>
+    <div class="col-xs-6">
```
Comment (adammeyer) [inline app/views/eventDetails/paymentOptions.html]: I think this looks better as xs-12, sm-6.

**Comment by OzzieOrca** [comment]
Hey @adammeyer how do you think this looks with the new filters?

Collapsed:
image
Opened:
image
xs devices:
image

**Comment by ryancarlson** [comment]
I like the concept a lot!

On Tue, Mar 3, 2015 at 1:26 AM, Scotty Waggoner <notifications@github.com>
wrote:

> Hey @adammeyer <https://github.com/adammeyer> how do you think this looks
> with the new filters?
>
> Collapsed:
> [image: image]
> <https://cloud.githubusercontent.com/assets/756501/6457443/bdaf51ec-c12a-11e4-90ae-19bcbedc8170.png>
> Opened:
> [image: image]
> <https://cloud.githubusercontent.com/assets/756501/6457439/af241112-c12a-11e4-956f-84e2cf214ed3.png>
> xs devices:
> [image: image]
> <https://cloud.githubusercontent.com/assets/756501/6457470/1707709e-c12b-11e4-939a-9690034585b8.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CruGlobal/conf-registration-web/pull/265#issuecomment-76893830>
> .
>



-- 
*Ryan Carlson*
Application Development Team
ryan.t.carlson@cru.org

**Comment by adammeyer** [comment]
Looks great scotty!

**Pull request merged**